wallet/src/Pos/Wallet/Web/Tracking/Sync.hs

```diff
@@ -446,6 +448,12 @@ applyModifierToWallet wid newTip CAccModifier{..} = do
     sortedAddedHistory <-
         getNewestFirst <$> sortWalletThByTime wid (DL.toList camAddedHistory)
     WS.updateHistoryCache wid $ sortedAddedHistory <> oldCachedHist
+    let cMetas = map (bimap encodeCType (CTxMeta . timestampToPosix)) $
+                 catMaybes $
+                    zipWith (\a b -> (a,) <$> b)
+                            (map _thTxId sortedAddedHistory)
+                            (map _thTimestamp sortedAddedHistory)
```

Comment:
It looks a bit complicated. You could use single `map` instead of two `map`s and one `zipWith`:
`map (\x -> (_thTxId x,) <$> _thTimestamp x) sortedAddedHistory`.